dev-docs/RFCs/v1.0/generic-edit-mode.md

```diff
@@ -0,0 +1,93 @@
+# Generic EditMode
+
+* **Author**: Clay Anderson
+
+## Summary
+
+Create a generic `EditMode` class in `@nebula/core` that is independent of deck.gl, react-map-gl, and GeoJSON. This generic class can then be integrated into `EditableGeoJsonLayer` as well as the [upcoming DrawControl feature of react-map-gl](https://github.com/uber/react-map-gl/issues/734)
+
+We will also refactor all the existing `ModeHandler` implementations of nebula to extend this class so that they can be used seamlessly between nebula.gl and react-map-gl.
+
+## Motivation
+
+There are two limitations with nebula's `ModeHandler` interface.
+
+1. It is dependent on deck.gl. This makes it unusable for `react-map-gl` which doesn't have a dependency on deck.gl.
+
+2. It is specific to GeoJSON. But there are desires to support editing other kinds of geometries (e.g. Hexagons using [H3](https://uber.github.io/h3/#/)).
+
+## API
+
+```javascript
+export type ModeState<TData, TGuides> = {
+  data: TData,
+  modeConfig: any,
+  selectedIndexes: number[],
+  guides: ?TGuides,
+  onEdit: (data: TData, editContext: any) => void
+  onUpdateGuides: (guides: ?TGuides) => void,
```

Comment:
Add more details / descriptions about these parameters.

Reply:
done. Please review and let me know if there are questions about them.

Reply:
What does `TGuides` look like?

Reply:
`TGuides` is intentionally abstract. For the BaseGeoJsonEditMode, it'll be a GeoJSON feature collection. But another type of edit mode (e.g. h3 hex editing) may use another data type for guides.